lib/zclient.c

```diff
@@ -1209,14 +1209,20 @@ int zapi_route_decode(struct stream *s, struct zapi_route *api)
 }
 
 bool zapi_route_notify_decode(struct stream *s, struct prefix *p,
+			      uint32_t *tableid,
 			      enum zapi_route_notify_owner *note)
 {
+	uint32_t t;
+
 	STREAM_GET(note, s, sizeof(*note));
 
 	STREAM_GETC(s, p->family);
 	STREAM_GETC(s, p->prefixlen);
 	STREAM_GET(&p->u.prefix, s,
-		   PSIZE(p->prefixlen));
+		   prefix_blen(p));
```

Comment:
Not really important, but I think it would be better to use `PSIZE` instead of `prefix_blen()` here and in the `zsend_route_notify_owner()` function. That's what we usually do when encoding/decoding prefixes because it saves us from encoding and decoding unnecessary bytes when the prefix length is small.